processor/logdedupprocessor/processor.go

```diff
@@ -113,6 +112,7 @@ func (p *logDedupProcessor) handleExportInterval(ctx context.Context) {
 			return
 		case <-ticker.C:
 			p.mux.Lock()
+			defer p.mux.Unlock()
```

Comment:
This lock/unlock is within an unbounded for loop, so deferring it will not have the desired effect. Should we move the contents of this case into a function so we can defer it?

Reply:
Moved in [59834c1](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34607/commits/59834c16d225eaa3e3e69a05c6269644261a2b05) 👍🏻